ci-operator/step-registry/ipi/conf/alibabacloud/ipi-conf-alibabacloud-commands.sh

```diff
@@ -5,14 +5,24 @@ set -o errexit
 set -o pipefail
 
 # copy the creds to the SHARED_DIR
-cp /var/run/vault/alibaba/alibabacreds.ini ${SHARED_DIR}
-cp /var/run/vault/alibaba/config ${SHARED_DIR}
-cp /var/run/vault/alibaba/envvars ${SHARED_DIR}
+if test -f "${CLUSTER_PROFILE_DIR}/alibabacreds.ini" 
```

Comment:
I don't see a big problem here but I'm curious to know what the difference is from the `CLUSTER_PROFILE_DIR` and  using `vault`. I would think we'd want the credentials to live in vault but maybe someone can educate me on the difference and the reasons to store the credentials with the `CLUSTER_PROFILE_DIR`.

Reply:
We followed up [doc](https://docs.ci.openshift.org/docs/how-tos/adding-a-cluster-profile/#what-is-a-cluster-profile), and think CLUSTER_PROFILE_DIR is easier to use, also noticed existing code using it, e.g. https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/install/install/ipi-install-install-commands.sh#L203-L225. Thanks!

Reply:
@jianlinliu Any idea? Thanks!

Reply:
Per my understanding, vault can be directly mounted as secret volume to a specific ref, also vault can be used to define `CLUSTER_PROFILE` (vault credentials will be saved in `CLUSTER_PROFILE_DIR`). This is more depending on what purpose of the vault key/value. If the vault credential is for cloudprovider account credential, it is varying per different `CLUSTER_PROFILE`, then it should be used to define `CLUSTER_PROFILE`, if directly mount it as secret volume for a ref step, user want to install the cluster in his/her `CLUSTER_PROFILE` account, then the mounted vault secret would be kinda of hardcode, can not be used to provision clusters in different cloudprovider account.